### PR TITLE
[ltm-1.6] Improve location detail view performance

### PR DESCRIPTION
**Pull request by Kircheneer**
# Closes #DNE
# What's Changed

This should significantly speed up

- the location detail view
- any views utilizing `dcim/inc/location_hierarchy.html`

by doing a couple of things:

- Using `without_tree_fields` on the queryset for the view to ensure that `get_object_or_404` in the `get` method of the view doesn't have to calculate the location tree (it doesn't use any of the fields the calculation adds
- Using `without_tree_fields` on the query for the children of the location for the same reasons
- Implement a `LocationQuerySet.ancestors_without_cte` that provides a more optimized implementation of `TreeQuerySet.ancestors` in cases where the tree is shallow but wide
- Use this method in a couple of places

# Screenshots

_With ~40k locations_

ltm-1.6 @ d024edb7060ad665baf62e7aec9dbf66ecb6bdd3:
Screenshot 2024-02-08 at 09 52 33

this branch:
Screenshot 2024-02-08 at 09 51 50

These numbers are pretty consistent when reloading the page. Note that these are number reproduced in a simple dev environment, the view in the actual deployment currently takes ~6-7s.

# TODO
- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design
- Further things to explore:
- [x] Pre-computing `ancestors` for `dcim/inc/location_hierarchy.html` because this still triggers a full tree query whenever used - we can optimize this if we assume our trees are shallow
- [ ] Applicability to 2.0
- [ ] Usage of `ancestors_without_cte` in other places

### CODE REVIEW

File: nautobot/dcim/models/locations.py [hunk 1]
```diff
@@ -125,6 +125,22 @@ def get_for_model(self, model):
         content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
         return self.filter(location_type__content_types=content_type)
 
+    @staticmethod
+    def ancestors_without_cte(of, *, include_self=True):
```
Comment (Kircheneer) [inline nautobot/dcim/models/locations.py:129]: - Try to merge this with the normal ancestors method
- Put this into the base `TreeQuerySet`

**Review by Kircheneer — COMMENTED** [review]
Discussion with @glennmatthews

**Comment by Kircheneer** [comment]
Updated to match #5255

File: nautobot/core/tests/test_tree_queries.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from nautobot.core.testing import TestCase
```
Comment (glennmatthews) [inline nautobot/core/tests/test_tree_queries.py:1]:
```suggestion
from nautobot.utilities.testing import TestCase
```

**Review by timizuoebideri1 — APPROVED** [review]

**Review by HanlinMiao — APPROVED** [review]

**Pull request merged**